# Basics of Python

### Variables

- Variables are reserved memory that store data
- Variables are containers for storing data values
- Variables do not need to be declared with any particular type, and can even change type after they have been set.
- Variable name cant be start with number or special character
- A variable name must start with a letter or the underscore character
- A variable name cannot start with a number
- A variable name can only contain alpha-numeric characters and underscores (A-z, 0-9, and _ )
- Variable names are case-sensitive (age, Age and AGE are three different variables)
- A variable name cannot be any of the Python keywords

print('basic of python')

In [27]:
name = 10;

In [28]:
type(name)

int

In [29]:
nam = 10.2

In [30]:
type(nam)

float

In [31]:
name1 = name*1.2

In [32]:
name1

12.0

In [34]:
a = 'vijay'

In [36]:
type(a)

str

In [44]:
3= 'vijay'

SyntaxError: cannot assign to literal (4017616921.py, line 1)

  File "C:\Users\MR.GODHADE\AppData\Local\Temp\ipykernel_4184\4017616921.py", line 1
    3= 'vijay'
    
    ^
SyntaxError: cannot assign to literal

### MULTIPLE VARIABLE ASSIGNMENT

- Many Values to Multiple Variables

In [39]:
a,b,c,d = 1,2,3,4

In [40]:
a

1

In [41]:
b

2

In [42]:
c

3

In [43]:
d

4

- One Value to Multiple Variables

In [45]:
x=y=z='vijay'

In [46]:
x

'vijay'

In [47]:
y

'vijay'

In [48]:
z

'vijay'

## Data_types

- str
- Numeric ->      int, float, complex
- Collection  ->  list, tuple,dictionary, set
- Boolean

In [49]:
result = 10>20

In [50]:
result

False

In [52]:
type(result)

bool

In [53]:
bool(10)   # anything apart from 0, getting True

True

In [54]:
bool(0)

False

In [55]:
bool(-1)

True

## Implicite type casting

it will change type automatically acording to need of opertion

In [56]:
v = 10
type(v)

int

In [57]:
v = v*2.0


In [58]:
v

20.0

In [59]:
type(v)

float